prow/external-plugins/cherrypicker/server.go

```diff
@@ -269,9 +270,25 @@ func (s *Server) handlePullRequest(l *logrus.Entry, pre github.PullRequestEvent)
 		}
 		requestorToComments[c.User.Login][targetBranch] = &c
 	}
+
 	if len(requestorToComments) == 0 {
-		return nil
+		// didn't find any magic comments so look for magic labels instead
```

Comment:
Why are you making the two mutually exclusive? In the current approach both should work or did I miss something?

What about changing the whole plugin to just create the label on comment and create cherrypick just based on the labels and not on comments?

Reply:
Good point, I updated things to make it so both the comments and labels can be used together.

As for changing the flow to have the comments create/set labels, and then just have the cherry pick respond to the labels, what do you think the benefits of that approach would be? It would change the existing visible behavior, potentially creating new labels on repos that may not want 'em.

Reply:
Yeah the arbitrary label handling would not be ideal. The motivation would be that I imagine the code to get more straightforwards, the cherrypick actuation can act on merge events only and then defer from the labels if there is something to do, the label add actuation can act on comment creation/edit only

Reply:
Given  the downside of creating labels where  none existed, I think it is wiser to stick with the current implementatioon.